apps/keyboard/js/imes/myanmar/myanmar.js

```diff
@@ -0,0 +1,249 @@
+      { num: 2, out: 2, t:1, p1: M.vowelIN, p2: M.vowelIU },
+      { num: 2, out: 2, t:1, p1: M.mediaWA, p2: M.mediaYR },
+      { num: 2, out: 2, t:1, p1: M.mediaHA, p2: M.mediaYW }
+  ];
+
+//==================================================================
+//  \u1025 + \u102E => \u1026
+//  \u1025 + \u103A => \u1009 + \u103A
+//  \u101E + \u103C + \u1031 + \u102C + \u103A => \u102A
+//==================================================================
+  var ADD_RULE = [
+      {num: 2, out: '\u1026', p1: M.vowelSU, p2: M.vowelDI},
+      {num: 2, out: '\u1009\u103A', p1: M.vowelSU, p2: M.vowelAT},
+      {num: 5, out: '\u102A', p1: M.consnSA, p2: M.mediaRA, p3: M.vowelSE, p4: M.vowelSA, p5: M.vowelAT}
+  ];
+
+//==================================================================
+//  \u1039 + consonT[*] + \u1031 + < VK_BACK > => \u1039 + $filler + \u1031
+//  \u200B + consonT[*] + \u1031 + < VK_BACK > => $filler + \u1031
+//  mediaYH[*] + \u1031 + < VK_BACK > => \u1031
+//  \u1039 + consonT[*] + < VK_BACK > => NULL
+//  \u200B + \u1031 + < VK_BACK > => NULL
+//==================================================================
+  var DEL_RULE = [
+      {num: 3, out: '\u1031\u1039', p1: M.vowelVR, p2: M.consonT, p3: M.vowelSE},
+      {num: 2, out: '\u200B\u1031', p1: M.consonT, p2: M.vowelSE},
+      {num: 2, out: '\u1031', p1: M.mediaYH, p2: M.vowelSE},
+      {num: 2, out: '', p1: M.vowelVR, p2: M.consonT},
+      {num: 2, out: '', p1: M.zeroLEN, p2: M.vowelSE}
+  ];
+
+  function myanParser(rule, callback) {
+    var idx = ['p0', 'p1', 'p2', 'p3', 'p4', 'p5'];
+    var len = buffer.length;
+
+    if (len < 2) {
+      return null;
+    }
+
+    for (var i = 0; i < rule.length; i++) {
+      if (len < rule[i].num) {
+         continue;
+      }
+      var c = len - 1;
+      for (var j = rule[i].num; j > 0; j--) {
+        if (!buffer[c--].match( rule[i][idx[j]] )) {
+          break;
+        }
+      }
+      if (j == 0) {
+        console.log('mykeyboard - myanparser hit : i(' + i + ')');
+        return callback(rule[i]);
+      }
+    }
+
+    return null;
+  }
+
+  function processBuffer(callback) {
+    var swap = myanParser(SWAP_RULE, function(entry) {
+        if (!entry) {
+          return null;
+        }
+        var len = buffer.length;
+        var tail = entry.t;
+        var out = entry.out;
+        var s = {};
+        s.num = entry.num;
+        s.str = buffer.substr(len-tail, tail) +
+                buffer.substr(len-out, out-tail);
+        return s;
+    });
+
+    // After process swap, check if string need to be mreged.
+    var adds = myanParser(ADD_RULE, function(entry) {
+          if (!entry) {
+            return null;
+          }
+          var s = {};
+        console.log('mykeyboard - adds(' + entry.out +')');
+          s.num = entry.num;
+          s.str = entry.out;
+          return s;
+    });
+
+    return (adds) ? callback(adds) : callback(swap);
+  }
+
+  function backspace(callback) {
```

Comment:
It seems you could just make `backspace()` function return dels, and then invoke the function you want after that.
I don't get what's the purpose we need to pass a function into it.